### PR TITLE
testing refactor using ginkgo and gomega

### PR DESCRIPTION
**Pull request by one-horned-flying**
use gomega and ginkgo as the testing framework

### CODE REVIEW

File: oxide.go [hunk 1]
```diff
@@ -481,11 +481,7 @@ func (d *Driver) SetConfigFromFlags(opts drivers.DriverOptions) error {
 		errRequiredFlag = errors.Join(errRequiredFlag, errors.New(errRequiredOptionNotSet+flagBootDiskImageID))
 	}
 
-	if errRequiredFlag != nil {
-		return errRequiredFlag
-	}
-
-	return nil
+	return errRequiredFlag
```
Comment (one-horned-flying) [inline oxide.go:484]: `errRequiredFlag` will be nil if there were no flag errors, so we shouldn't need the nil check and return nil

File: oxide_test.go [hunk 2]
```diff
@@ -6,79 +6,95 @@
...
+			//flagProject:                   "",
+			//flagVCPUs:                     1,
+			//flagMemory:                    1024,
+			//flagBootDiskSize:              20,
+			//flagBootDiskImageID:           "",
+			//flagVPC:                       "",
+			//flagSubnet:                    "",
+			//flagUserDataFile:              "",
+			//flagSSHUser:                   "",
+			//flagAdditionalSSHPublicKeyIDs: []string{},
```
Comment (sudomateo) [inline oxide_test.go:90]: What's the purpose for these commented out items?

**Review by sudomateo — COMMENTED** [review]
Thank you for working on this. I haven't used `ginkgo` before but the tests pass.

```sh
> ginkgo
Running Suite: RancherMachineDriverOxide Suite - /home/sudomateo/projects/rancher-machine-driver-oxide
======================================================================================================
Random Seed: 1734639120

Will run 17 of 17 specs
•••••••••••••••••

Ran 17 of 17 Specs in 0.001 seconds
SUCCESS! -- 17 Passed | 0 Failed | 0 Pending | 0 Skipped
PASS

Ginkgo ran 1 suite in 418.89044ms
Test Suite Passed
```

Left a few comments and a question before merging this.

File: oxide_test.go (hunk 2 above)
Comment (one-horned-flying) [reply oxide_test.go:90]: removing those.. not needed as I thought

**Comment by one-horned-flying** [comment]
ginkgo and gomega are the standard for https://www.kubebuilder.io and I'm starting to use it more from building out an operator.

**Review by sudomateo — APPROVED** [review]

**Pull request merged**